common/persistence/elasticsearch/processor.go

```diff
@@ -368,9 +381,23 @@ func extractErrorReason(resp *elastic.BulkResponseItem) string {
 	return ""
 }
 
-func newAckChanWithStopwatch(ackCh chan<- bool, stopwatch metrics.Stopwatch) *ackChanWithStopwatch {
-	return &ackChanWithStopwatch{
-		ackCh:                 ackCh,
-		addToProcessStopwatch: stopwatch,
+func newAckChan() *ackChan {
+	return &ackChan{
+		ackChInternal: make(chan bool, 1),
+		addedAt:       time.Now().UTC(),
+	}
+}
+
+func (a *ackChan) Start(metricsClient metrics.Client) {
+	metricsClient.RecordTimer(metrics.ElasticsearchBulkProcessor, metrics.ElasticsearchBulkProcessorWaitLatency, time.Now().UTC().Sub(a.addedAt))
+	a.startedAt = time.Now().UTC()
+}
+
+func (a *ackChan) Done(ack bool, metricsClient metrics.Client) {
+	a.ackChInternal <- ack
```

Comment:
any possibility this channel is already acked or what if `Done` is called > 1 times

Reply:
above code seem to have possibility to get the the channel > 1 times, so `a.ackChInternal <- ack` should be 
```go
select {
case a.ackChInternal <- ack:
default:
    // noop
}
```

Reply:
This is just a refactoring. This code was there just wasn't extracted as a method. There are 2 calls to `Done()` in code above: normal case (success/error) or duplicate `visibilityTaskKey`. Both of them a guaranteed to be called only once by `mapToAckChan` and in case of duplicate the item get overwritten and never used again.